core/bentley/src/BentleyError.ts

```diff
@@ -8,6 +8,69 @@
 
 import { DbResult } from "./BeSQLite";
 import { RepositoryStatus } from "./internal/RepositoryStatus";
+import { Optional } from "./UtilityTypes";
+
+type BaseError = Error;
+
+/**
+ * Namespace for handling all exceptions thrown by iTwin.js
+ * @beta
+ */
+export namespace ITwinError {
```

Comment:
nit:

Do we really need a namespace here? I know our codebase is littered with namespaces so it will be impossible to completely get away from it anytime soon, but we have been trying to move away from adding any new namespaces.

We could move all the new additions to a new module called "ITwinError.ts". We could then also eliminate the type alias BaseError, and the interface currently defined as Error can be renamed to ITwinError. Then all utility functions (createError, throwError, isError) could be renamed to include ITwin in it; while a bit more verbose it is (imo) a cleaner way than using a namespace. The isObject type guard could be extracted out of the namespace, and could be used more generically than just in the context of ITwinErrors

Reply:
Namespaces impede tree-shaking. OTOH, free functions are less discoverable than a group of related functions aggregated into a namespace. Do you have other reasons to dislike namespaces?

Reply:
With namespaces, we'll have:
- `ITwinError.isError`
- `BentleyError.isError`
- `ChannelError.isError`
- etc.

When others create their own set of Errors, they'll create a namespace for it following the pattern in `ChannelError`. Seems very clean to me. Otherwise there'll be just lots of loose top-level types related only by some prefix. I get tree shaking, but that's true of every class too,

Reply:
@aruniverse I pulled the interfaces out of the namespaces. Do you have further objections/suggestions?